app/assets/javascripts/tree_display.jsx

```diff
@@ -633,561 +640,562 @@ jQuery(document).ready(function() {
+                        jQuery.each(data2, function(nodeType, outerNode) {
+                            jQuery.each(outerNode, function(i, node) {
+                                var newParams = {
+                                    key: node.name + "|" + node.directory,
+                                    nodeType: nodeType,
+                                    child_nodes: node.nodeinfo
+                                }
+                                if (nodeType === 'Assignments') {
+                                    node["children"] = null;
+                                    node[newParams]=newParams;
+                                } else if (nodeType === 'Courses') {
+                                    newParams["nodeType"] = 'CourseNode'
+                                    node["newParams"]=newParams;
+                                } else if (nodeType === 'Questionnaires') {
+                                    newParams["nodeType"] = 'FolderNode'
+                                    node["newParams"]=newParams;
+                                }
+                            })
+                        })
+                        if (data2) {
+                            _this.setState({
+                                tableContent: data2
+                            })
+                        }
+                    },
+                    'json')
+            })
 
-  if (document.getElementById("tree_display")) {
-    React.render(
-      React.createElement(TabSystem),
-      document.getElementById("tree_display")
-    )
-  }
+        },
+        handleTabChange: function(tabIndex) {
+            jQuery.get("/tree_display/set_session_last_open_tab?tab="+tabIndex.toString())
+        },
+        render: function() {
+            return (
+                <ReactSimpleTabs
+                    className="tab-system"
+                    tabActive={parseInt(this.state.activeTab)}
+                    onAfterChange={this.handleTabChange}
+                >
+                    <ReactSimpleTabs.Panel title="Courses">
+                        <FilterableTable key="table1" dataType='course' data={this.state.tableContent.Courses}/>
+                    </ReactSimpleTabs.Panel>
+                    <ReactSimpleTabs.Panel title="Assignments">
+                        <FilterableTable key="table2" dataType='assignment' data={this.state.tableContent.Assignments}/>
+                    </ReactSimpleTabs.Panel>
+                    <ReactSimpleTabs.Panel title="Questionnaires">
+                        <FilterableTable key="table2" dataType='questionnaire' data={this.state.tableContent.Questionnaires}/>
+                    </ReactSimpleTabs.Panel>
+                </ReactSimpleTabs>
+            )
+        }
+    })
+
+    if (document.getElementById("tree_display")) {
+        React.render(
+            React.createElement(TabSystem),
+            document.getElementById("tree_display")
+        )
+    }
 
 })
```

Comment:
Seems like you did not intend to change this file but introduced many whitespaces and newlines unknowingly. Please correct it. (remove this file from commit if it does not have any changes relevant to your project  or correct its code climate )